scipy/stats/tests/test_odds_ratio.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal, assert_allclose
+from scipy.stats import nchypergeom_fisher, hypergeom
+from scipy.stats._odds_ratio import odds_ratio
+from .fisher_exact_results_from_r import data
+
+
+class TestOddsRatio:
+
+    @pytest.mark.parametrize('parameters, rresult', data)
+    def test_results_from_r(self, parameters, rresult):
+        alternative = parameters.alternative.replace('.', '-')
+        result = odds_ratio(parameters.table, alternative=alternative)
+        # The results computed by R are not very precise.
+        if result.odds_ratio < 400:
+            or_rtol = 5e-4
+            ci_rtol = 2e-2
+        else:
+            or_rtol = 5e-2
+            ci_rtol = 1e-1
+        assert_allclose(result.odds_ratio,
+                        rresult.conditional_odds_ratio, rtol=or_rtol)
+        assert_allclose(result.pvalue, rresult.pvalue, rtol=1e-6)
+        ci = result.odds_ratio_ci(parameters.confidence_level)
+        assert_allclose((ci.low, ci.high), rresult.conditional_odds_ratio_ci,
+                        rtol=ci_rtol)
+
+        # Also do a self-check for the conditional odds ratio.
+        # With the computed conditional odds ratio as the noncentrality
+        # parameter of the noncentral hypergeometric distribution with
+        # parameters table.sum(), table[0].sum(), and table[:,0].sum() as
+        # total, ngood and nsample, respectively, the mean of the distribution
+        # should equal table[0, 0].
+        cor = result.odds_ratio
+        table = np.array(parameters.table)
+        total = table.sum()
+        ngood = table[0].sum()
+        nsample = table[:, 0].sum()
+        # nchypergeom_fisher does not allow the edge cases where the
+        # noncentrality parameter is 0 or inf, so handle those values
+        # separately here.
```

Comment:
Should we fix that?

Reply:
I don't think we have to, and I'm not sure we should.  Technically, if the odds ratio is 0 or inf, the support collapses to a single value.  So would we also modify the calculation of the support of the nchg distributions in these edge cases?  I think we can defer such a discussion to another time.

Reply:
OK.